From e7431385cbd6a5f5dc5d5b38e25014c8d15842dd Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 19 May 2015 16:37:22 +0100
Subject: [PATCH] p2v: Explicitly depend on nm-connection-editor.

Previously it was only being pulled in implicitly.  However virt-p2v
runs this binary from the "Configure Network" button.

This fixes commit e464774a79ff9eeedcdb3fd53efd5a7898ca036e.

(cherry picked from commit f6dd8a3221cfa7b5aa6434cddcd0d636dd64e02b)
---
 p2v/p2v.ks.in             | 1 +
 p2v/virt-p2v-make-disk.in | 9 +++++----
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/p2v/p2v.ks.in b/p2v/p2v.ks.in
index 2319781..58cca5e 100644
--- a/p2v/p2v.ks.in
+++ b/p2v/p2v.ks.in
@@ -70,6 +70,7 @@ metacity
 pcre
 libxml2
 gtk2
+nm-connection-editor
 network-manager-applet
 dbus-x11
 hwdata
diff --git a/p2v/virt-p2v-make-disk.in b/p2v/virt-p2v-make-disk.in
index 6d7dcb2..28d6e75 100644
--- a/p2v/virt-p2v-make-disk.in
+++ b/p2v/virt-p2v-make-disk.in
@@ -96,13 +96,14 @@ trap cleanup INT QUIT TERM EXIT ERR
 #   - hardware support (firmware etc, RHBZ#1157679)
 #   - metacity (window manager, another could be used)
 #   - NetworkManager
+#   - nm-connection-editor
 #   - nm-applet
 #   - dbus-x11 (required by nm-applet, but not specified as a dep in Fedora)
 #
 # Note that libguestfs is NOT a dependency.
 case "$osversion" in
     centos-*|fedora-*|rhel-*|scientificlinux-*)
-        deps=pcre,libxml2,gtk2,/usr/bin/xinit,/usr/bin/ssh,/usr/bin/qemu-nbd,/usr/bin/Xorg,xorg-x11-drivers,xorg-x11-fonts-Type1,metacity,NetworkManager,network-manager-applet,dbus-x11,hwdata,@hardware-support
+        deps=pcre,libxml2,gtk2,/usr/bin/xinit,/usr/bin/ssh,/usr/bin/qemu-nbd,/usr/bin/Xorg,xorg-x11-drivers,xorg-x11-fonts-Type1,metacity,NetworkManager,nm-connection-editor,network-manager-applet,dbus-x11,hwdata,@hardware-support
         cat > $tmpdir/p2v.conf <<'EOF'
 add_drivers+=" usb-storage "
 EOF
@@ -120,13 +121,13 @@ EOF
         "
         ;;
     debian-*|ubuntu-*)
-        deps=libpcre3,libxml2,libgtk2.0-0,openssh-client,qemu-utils,xorg,xserver-xorg-video-all,metacity,network-manager,network-manager-applet,dbus-x11,hwdata
+        deps=libpcre3,libxml2,libgtk2.0-0,openssh-client,qemu-utils,xorg,xserver-xorg-video-all,metacity,network-manager,network-manager-gnome,network-manager-applet,dbus-x11,hwdata
         ;;
     archlinux-*)
-        deps=pcre,libxml2,gtk2,openssh,qemu,xorg-xinit,xorg-server,xf86-video-*,metacity,NetworkManager,network-manager-applet,dbus-x11,hwdata
+        deps=pcre,libxml2,gtk2,openssh,qemu,xorg-xinit,xorg-server,xf86-video-*,metacity,NetworkManager,nm-connection-editor,network-manager-applet,dbus-x11,hwdata
         ;;
     opensuse-*|suse-*)
-        deps=pcre,libxml2,gtk2,/usr/bin/ssh,/usr/bin/qemu-nbd,/usr/bin/xinit,/usr/bin/Xorg,xf86-video-*,metacity,NetworkManager,network-manager-applet,dbus-x11,hwdata
+        deps=pcre,libxml2,gtk2,/usr/bin/ssh,/usr/bin/qemu-nbd,/usr/bin/xinit,/usr/bin/Xorg,xf86-video-*,metacity,NetworkManager,nm-connection-editor,network-manager-applet,dbus-x11,hwdata
         ;;
     *)
         echo "$program: internal error: could not work out the Linux distro from '$osversion'"
-- 
1.8.3.1

